frontend/src/app/signUp/Organization/OrganizationForm.tsx

```diff
@@ -0,0 +1,223 @@
+
+import {
+  initOrg,
+  initOrgErrors,
+  organizationFields,
+  OrganizationTypeEnum,
+  organizationSchema as schema,
+} from "./utils";
+import { Thanks } from "./Thanks";
+
+import "./OrganizationForm.scss";
+
+export interface OrganizationCreateRequest {
+  name: string;
+  type: OrganizationType | "";
+  state: string;
+  firstName: string;
+  middleName?: string | null;
+  lastName: string;
+  email: string;
+  workPhoneNumber: string;
+}
+
+export interface OrganizationCreateResponse {
+  orgExternalId: string;
+}
+
+type OrganizationFormErrors = Record<keyof OrganizationCreateRequest, string>;
+
+const OrganizationForm = () => {
+  const [organization, setOrganization] = useState<OrganizationCreateRequest>(
+    initOrg()
+  );
+  const [errors, setErrors] = useState<OrganizationFormErrors>(initOrgErrors());
+
+  const [loading, setLoading] = useState(false);
+  const [formChanged, setFormChanged] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+
+  const onDetailChange = (field: keyof OrganizationCreateRequest) => (
+    value: OrganizationCreateRequest[typeof field]
+  ) => {
+    setFormChanged(true);
+    setOrganization({ ...organization, [field]: value });
+  };
+
+  const validateField = async (field: keyof OrganizationCreateRequest) => {
+    setErrors(
+      await isFieldValid({ data: organization, schema, errors, field })
+    );
+  };
+
+  const getValidationStatus = (field: keyof OrganizationCreateRequest) =>
+    errors[field] ? "error" : undefined;
+
+  const onSave = async () => {
+    setLoading(true);
+    const validation = await isFormValid({
+      data: organization,
+      schema,
+    });
+    if (validation.valid) {
+      try {
+        await SignUpApi.createOrganization(organization);
+      } catch (error) {
+        const alert = (
+          <Alert type="error" title="Submission Error" body={error} />
+        );
+        showNotification(toast, alert);
+        setLoading(false);
+        return;
+      }
+      setErrors(initOrgErrors());
+      setSubmitted(true);
+      return;
+    }
+    setErrors(validation.errors);
+    const alert = (
+      <Alert
+        type="error"
+        title="Form Errors"
+        body="Please check the form to make sure you complete all of the required fields."
+      />
+    );
+    showNotification(toast, alert);
+    setLoading(false);
+  };
+
+  if (submitted) {
+    return <Thanks />;
+  }
+
+  if (loading) {
+    return <LoadingCard message="Creating organization" />;
+  }
+
+  const getFormElement = (
+    field: keyof OrganizationCreateRequest,
+    label: string | React.ReactNode,
+    required: boolean,
+    subheader: string | null
+  ) => {
+    switch (field) {
+      case "state":
+        return (
+          <Select
+            label={label}
+            name="state"
+            value={organization.state || ""}
+            options={stateCodes.map((c) => ({ label: c, value: c }))}
```

Comment:
**product question:** Should we have the dropdown only include valid states, instead of all states?

@vz3 @LGFranklin

Reply:
Note: I think we'll transition to only valid states once we get Victor's preferred "choose your state" modal up so we can appropriately redirect to waitlist/account request, but I'm wondering if we should only show valid states in the interim

Reply:
It shows all states, but if you choose an invalid state, it pops up an error message with a link to the page on the static site about current valid jurisdictions!

Reply:
Thanks! That's probably good for now. I feel like I saw a ticket to restrict the dropdown to only valid states but can't find it anymore 🤔